### PR TITLE
Resolve `sync` only after heads have been added

### PR DESCRIPTION
**Pull request by tyleryasaka**
This is required for a tentative solution to https://github.com/orbitdb/orbit-db/issues/509:

https://github.com/orbitdb/orbit-db/pull/514

See [this commit message](https://github.com/orbitdb/orbit-db-store/pull/38/commits/ddac2ad7dbd4e57c1045237255b653eda10d8910) for a more detailed explanation.

### CODE REVIEW

**Comment by aphelionz** [comment]
@chmanie Thank you for the clarification. Can you and @tyleryasaka work to get the event listeners cleaned up?

File: src/Store.js [hunk 1]
```diff
@@ -259,10 +259,19 @@ class Store {
       return head
     }
 
-    return mapSeries(heads, saveToIpfs)
-      .then(async (saved) => {
-        return this._replicator.load(saved.filter(e => e !== null))
+    const saved = await mapSeries(heads, saveToIpfs)
+    await this._replicator.load(saved.filter(e => e !== null))
+
+    return new Promise(resolve => {
```
Comment (chmanie) [inline src/Store.js]: I feel like this also needs a timeout to be rejected at some point

**Comment by oed** [comment]
So basically just removing the event listeners before resolving the promise? @chmanie

File: src/Store.js (hunk 1 above)
Comment (oed) [reply src/Store.js]: Not sure I follow, why would you want to do that?

**Comment by aphelionz** [comment]
Friendly ping on this thread @chmanie @oed

**Comment by chmanie** [comment]
> So basically just removing the event listeners before resolving the promise

Yeah, that should do.

File: src/Store.js (hunk 1 above)
Comment (chmanie) [reply src/Store.js]: `resolve` is wrapped in conditionals both of the times. Are we sure one of those is always going to happen?

Comment (oed) [reply src/Store.js]: I'm pretty sure yes. If not something with the replication logic would be broken

**Comment by aphelionz** [comment]
@tyleryasaka Would you be able to remove those event listeners? If so then we can merge this and the commensurate PR in `orbit-db`

**Pull request merged**